presto-main/src/main/java/com/facebook/presto/operator/Operator.java

```diff
@@ -52,7 +52,7 @@
      */
     default ListenableFuture<?> isBlocked()
     {
-        return NOT_BLOCKED;
+        return getOperatorContext().isWaitingForMemory();
```

Comment:
Because some of them, like partial aggregation ignore `isWaitingForMemory()` because they're going to flush their memory usage. Modifying `Driver` is the approach I tried to take first though, so I'll spend some more time thinking about how to make it work